composeApp/src/commonMain/kotlin/org/ooni/engine/models/NetworkType.kt

```diff
@@ -0,0 +1,33 @@
+package org.ooni.engine.models
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+
+@Serializable(with = NetworkTypeSerializer::class)
+enum class NetworkType(val value: String) {
+    VPN("vpn"),
+    Wifi("wifi"),
+    Mobile("mobile"),
```

Comment:
We may want to start thinking of devices connected to the internet through `TRANSPORT_BLUETOOTH`, `TRANSPORT_ETHERNET`, `TRANSPORT_USB` especially on android that supports these options in the *Mobile Hotspot and Tethering* option.

Reply:
Interesting. Didn't think about those, I was just following the current implementation on the Android app. Maybe that will be future work, after we reach feature parity?

Reply:
yeah that works for me.